ivy_tests/test_ivy/test_frontends/test_numpy/test_creation_routines/test_building_matrices.py

```diff
@@ -161,3 +163,33 @@ def test_numpy_vander(
         N=N,
         increasing=increasing,
     )
+
+
+# diagflat
+@handle_frontend_test(
+    fn_tree="numpy.diagflat",
+    args_packet=_generate_diag_args(),
+    test_with_out=st.just(False),
+)
+def test_numpy_diagflat(
+    *,
+    frontend,
+    test_flags,
+    fn_tree,
+    on_device,
+    args_packet,
+):
+    dtype_x, offset, dtype_padding_value, align, num_rows, num_cols = args_packet
```

Comment:
the full test body can be simplified to:
```
dtype, x, k = dtype_and_x_k
    helpers.test_frontend_function(
        input_dtypes=dtype,
        frontend=frontend,
        test_flags=test_flags,
        fn_tree=fn_tree,
        on_device=on_device,
        v=x[0],
        k=k,
    )
```

Reply:
Thanks